src/parcsr_ls/ams.c

```diff
@@ -2597,21 +2589,15 @@ hypre_AMSComputeGPi(hypre_ParCSRMatrix *A,
       HYPRE_Real *Gx_data, *Gy_data = NULL, *Gz_data = NULL;
 
       MPI_Comm comm = hypre_ParCSRMatrixComm(G);
+      HYPRE_BigInt *col_starts_G = hypre_ParCSRMatrixColStarts(G);
       HYPRE_BigInt global_num_rows = hypre_ParCSRMatrixGlobalNumRows(G);
       HYPRE_BigInt global_num_cols = dim * hypre_ParCSRMatrixGlobalNumCols(G);
       HYPRE_BigInt *row_starts = hypre_ParCSRMatrixRowStarts(G);
-      HYPRE_BigInt *col_starts;
-      HYPRE_Int col_starts_size;
+      HYPRE_BigInt col_starts[2] = {(HYPRE_BigInt)dim * col_starts_G[0],
```

Comment:
@rfalgout There was a memory leak here. `col_starts` was not being freed. I simplified this to use a static array. (CMake helped finding this with the Wextra flag)

Reply:
Nice!